src/modules/nodeClient/NodeClient.ts

```diff
@@ -0,0 +1,98 @@
+import axios, { AxiosInstance, AxiosResponse } from "axios";
+import { handleNetworkError } from "boa-sdk-ts";
+import { URL } from "url";
+import { INodeInformation } from "../../Types";
+import moment from "moment";
+
+export class NodeClient {
+    /**
+     * The network endpoint to connect to Node
+     */
+    private endpoint: URL;
+
+    /**
+     * The instance of the axios
+     */
+    private client: AxiosInstance;
+
+    constructor(endpoint: URL) {
+        this.endpoint = endpoint;
+        this.client = axios.create({
+            baseURL: endpoint.toString(),
+            // Timeout is in ms
+            timeout: 10000,
```

Comment:
Isn't this timeout causing delays in all API request responses for STOA?

Reply:
As like other services this will not effect the stoa APIs